spec/classes/puppet_server_puppetserver_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'spec_helper'
+
+describe 'puppet::server::puppetserver' do
+  on_supported_os.each do |os, facts|
+    let(:facts) do
+      facts.merge({
+        :concat_basedir => '/foo/bar',
+        :puppetversion  => Puppet.version,
+      })
+    end
+
+    describe 'with default parameters' do
+      let :pre_condition do
+        "class {'puppet':
```

Comment:
I think you could test puppet::server::puppetserver by itself (since everything you need is parameterised), which will help keep our tests fast.